src/System.Net.Security/src/System/Net/SecureProtocols/SslState.cs

```diff
@@ -596,9 +596,17 @@ internal void ProcessAuthentication(LazyAsyncResult lazyResult)
                         KeyExchangeStrength);
                 }
             }
+            catch (Exception)
+            {
+                // If an exception emerges synchronously, the asynchronous operation was not
+                // initiated, so no operation is in progress.
+                _nestedAuth = 0;
```

Comment:
Writes of Int32s are atomic in that they're not subject to torn reads/writes.  Interlocked.Exchange provides atomicity for not only writing the value but also reading the current value out as part of the atomic operation, but we don't need to do the check here.  Technically the Exchange could probably be removed from the EndXx method (if its purpose were to catch incorrect parallel usage then it would be necessary, and likely more so as a CompareExchange), but I don't want to mess with it.